pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/HashUtils.java

```diff
@@ -36,6 +40,35 @@ public static byte[] hashMD5(byte[] bytes) {
     return Hashing.md5().hashBytes(bytes).asBytes();
   }
 
+  /**
+   * For use-cases where the primary-key is set to columns that are guaranteed to be type-4 UUIDs, this hash-function
+   * will reduce the number of bytes required from 36 to 16 for each UUID, without losing any precision. This leverages
+   * the fact that a type-4 UUID is essentially a 16-byte value.
+   */
+  public static byte[] hashUUIDv4(byte[] bytes) {
```

Comment:
Consider picking a more explicit name for type-4 (is type-4 an official name for this kind of UUID?)? V4 is a little bit weird.

Reply:
Yeah I'll update this. The function should be able to handle other variants of UUID since all of them have 128 bits. These are defined in and also well defined in UUID Javadocs: https://www.ietf.org/rfc/rfc4122.txt

Reply:
Updated. The function should be able to handle any UUID conformant to RFC 4122 (Java's UUID is conformant to the same). Marking PR as ready for review. Lmk your thoughts.